fix docs: add guide for update docs (#1359)
oam-dev/kubevela: docs/README.md

> # KubeVela Docs

Reviewer: ```suggestion
# Contributing to KubeVela Docs 
```

> [Kubevela website](http://kubevela.io/)  is built using [Docusaurus 2](https://v2.docusaurus.io/), a modern static website generator.

Reviewer: ```suggestion
Here is the source code of [Kubevela website](http://kubevela.io/). It's built by [Docusaurus 2](https://v2.docusaurus.io/), a modern static website generator.
```

> Whenever you modify the files under docs and submit a PR, the `check-docs` action will verify whether your docs are accurate.

Reviewer: ```suggestion
Any files modifid here wil trigger the `check-docs` Github action to run and validate the docs could be build successfully into the website.
Please you can build and preview it locally.
```

> After PR is merged, files(`docs/en/*`, `resource/*`, `sidebars.js`, `docusaurus.config.js`, `index.js`) will be submitted to the corresponding locations in the repo 

Reviewer: ```suggestion
Any changes on these files(`docs/en/*`, `resource/*`, `sidebars.js`, `docusaurus.config.js`, `index.js`) will be submitted to the corresponding locations of the repo 
```

> [kubevela.io](https://github.com/oam-dev/kubevela.io). Finally, github-action will parse the document and publish it to the Kubevela website. 

Reviewer: ```suggestion
[kubevela.io](https://github.com/oam-dev/kubevela.io). The Github-Action there will parse the document and publish it to the Kubevela Website automatically. 
```

> This document will guide you how to update the KubeVela docs.

Reviewer: ```suggestion
Please follow our guides below to learn how to write the docs in the right way.
```

## Add or Update Docs

> When you add or update docs, you maybe need to fix 3 files: `docs/en/`, `resource/` and `sidebars.js`.

Reviewer: ```suggestion
When you add or modify the docs,  these three files(`docs/en/`, `resource/` and `sidebars.js`) should be taken into consideration.
```

> 1. `docs/en/`, markdown files are placed in this folder, every markdown file needs to 

Reviewer: ```suggestion
1. `docs/en/`, the main English documentation files are mainly located in this folder. All markdown files need to follow the format, 
```

>    add title at the beginning in the following format:

Reviewer: ```suggestion
   that the title at the beginning should be in the following format:
```

   
    ```markdown
    ---
    title: Title Name
    ---

    ```

>    When you want to add a link refer to `md` file inside `docs/en`, you need to use relative path and remove the `md` suffix.

Reviewer: ```suggestion
   When you want to add a link refer to any `.md` files inside the docs(`docs/en`), you need to use relative path and remove the `.md` suffix.
```

>    such as `en/helm/component.md` refer to `en/platform-engineers/definition-and-templates.md`. The link should like:

Reviewer: ```suggestion
   For example, the `en/helm/component.md` has a link refer to `en/platform-engineers/definition-and-templates.md`. Then the format should like:
```

   
    ```markdown
    [the definition and template concepts](../platform-engineers/definition-and-templates)
    ```
   

> 2. `resource/`, image files are placed in this folder, When you want to use image in md, please also use relative path:

Reviewer: ```suggestion
2. `resource/`, image files are located in this folder. When you want to use link any image in documentation, you should put the image resources here and use a relative path like below:
```

  
   ```markdown
    ![alt](../resources/concepts.png)
    ```

> 3. `sidebars.js`, this file contain the navigation information of the KubeVela docs.

Reviewer: ```suggestion
3. `sidebars.js`, this file contain the navigation information of the KubeVela website.
```

>     Please read [doc](https://docusaurus.io/docs/sidebar) to learn how to write sidebar.js.

Reviewer: ```suggestion
    Please read [the official docs of docusaurus](https://docusaurus.io/docs/sidebar) to learn how to write `sidebar.js`.
```

    
    ```js
       {
          type: 'category',
          label: 'Capability References',
          items: [

>             // This is the path under "docs/end" 

Author: "docs/end" to "docs/en"

>     ```js
>        {
>           type: 'category',
>           label: 'Capability References',
>           items: [
>             // This is the path under "docs/end" 
>             'developers/references/README',
>             'developers/references/workload-types/webservice',
>             'developers/references/workload-types/task',
>             ...
>           ],
>         },
>         {
>           type: 'category',
>           label: 'Roadmap',
>           items: [
>             'roadmap',
>           ],
>         },
>         {
>           type: 'doc',
>           id: 'developers/references/devex/faq'
>         }
>     ```

Reviewer: ```suggestion
```

We don't need to add example here if we're not explaining it here

>    

Reviewer: TODO: ADD how to translate into Chinese or other language here.

## Local Development

> You can develop website in local, but first install node and yarn.

Reviewer: ```suggestion
You can develop website locally with the `node` and `yarn` installed.
```

Please add a link about how to install them

> Every time you modify the files under docs, you need to re-run the following cmd:

Reviewer: ```suggestion
Every time you modify the files under the docs, you need to re-run the following command, it will not sync automatically:
```

```shell
make docs-start
```

## Build in Local

> You can build Kubevela website in local to test the correctness of docs, only run the following cmd:

Reviewer: ```suggestion
You can also preview the KubeVela website locally by:
```

Author: `make docs-build` can only build to test the correctness of docs
`make docs-start` can preview the KubeVela website locally
